admin_pages/events/Events_Admin_Page.core.php

```diff
@@ -2062,37 +2092,19 @@ private function _change_event_status($EVT_ID = 0, $event_status = '')
      * @access protected
      * @param bool $redirect_after
      */
-    protected function _delete_event($redirect_after = true)
+    protected function _delete_event()
     {
         // determine the event id and set to array.
         $EVT_ID = isset($this->_req_data['EVT_ID']) ? absint($this->_req_data['EVT_ID']) : null;
-        $EVT_ID = isset($this->_req_data['post']) ? absint($this->_req_data['post']) : $EVT_ID;
-        // loop thru events
-        if ($EVT_ID) {
-            $success = $this->_permanently_delete_event($EVT_ID);
-            // get list of events with no prices
-            $espresso_no_ticket_prices = get_option('ee_no_ticket_prices', array());
-            // remove this event from the list of events with no prices
-            if (isset($espresso_no_ticket_prices[ $EVT_ID ])) {
-                unset($espresso_no_ticket_prices[ $EVT_ID ]);
-            }
-            update_option('ee_no_ticket_prices', $espresso_no_ticket_prices);
-        } else {
-            $success = false;
-            $msg = esc_html__(
-                'An error occurred. An event could not be deleted because a valid event ID was not not supplied.',
-                'event_espresso'
-            );
-            EE_Error::add_error($msg, __FILE__, __FUNCTION__, __LINE__);
-        }
-        if ($redirect_after) {
-            $this->_redirect_after_action(
-                $success,
-                'Event',
-                'deleted',
-                array('action' => 'default', 'status' => 'trash')
-            );
-        }
+        wp_safe_redirect(
+            EE_Admin_Page::add_query_args_and_nonce(
+                [
+                    'action' => 'preview_deletion',
+                    'EVT_IDs[]' => $EVT_ID
+                ],
+                $this->_admin_base_url
+            )
+        );
```

Comment:
Oh yeah you're right, but I left it as-is because this is where we would use the batch system to generate the contents for the preview page. So right now it doesn't make sense. I'll add a comment explaining that.